tests/Aspire.Hosting.Oracle.Tests/OracleFunctionalTests.cs

```diff
@@ -0,0 +1,377 @@
+
+public class OracleFunctionalTests(ITestOutputHelper testOutputHelper)
+{
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyOracleResource()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(10));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 20, Delay = TimeSpan.FromSeconds(10) })
+            .Build();
+
+        var builder = CreateDistributedApplicationBuilder();
+
+        var oracleDbName = "freepdb1";
+
+        var oracle = builder.AddOracle("oracle");
+        var db = oracle.AddDatabase(oracleDbName);
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{db.Resource.Name}"] = await db.Resource.ConnectionStringExpression.GetValueAsync(default)
+        });
+
+        hb.AddOracleDatabaseDbContext<TestDbContext>(db.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        // Wait until the database is available
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var dbContext = host.Services.GetRequiredService<TestDbContext>();
+            var databaseCreator = (RelationalDatabaseCreator)dbContext.Database.GetService<IDatabaseCreator>();
+            Assert.True(await databaseCreator.CanConnectAsync(token));
+        }, cts.Token);
+
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var dbContext = host.Services.GetRequiredService<TestDbContext>();
+
+            var results = await dbContext.Database.ExecuteSqlRawAsync("SELECT 1", token);
+
+            Assert.Equal(1, results);
+        }, cts.Token);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false, Skip = "When using p@ssw0rd1 it fails with: Password cannot be null. Enter password: \n Then when using a different password the image takes 5 minutes to be ready.")]
+    [RequiresDocker]
+    public async Task WithDataShouldPersistStateBetweenUsages(bool useVolume)
+    {
+        var oracleDbName = "freepdb1";
+
+        string? volumeName = null;
+        string? bindMountPath = null;
+
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(10));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 20, Delay = TimeSpan.FromSeconds(10) })
+            .Build();
+
+        try
+        {
+            var builder1 = CreateDistributedApplicationBuilder();
+
+            var password = ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder1, "orcl-password").Value;
+
+            var passwordParameter = builder1.AddParameter("pwd");
+            builder1.Configuration["Parameters:pwd"] = password;
+
+            var oracle1 = builder1.AddOracle("oracle", passwordParameter);
```

Comment:
There's a slightly simpler pattern we can use:

```suggestion
            var oracle1 = builder1.AddOracle("oracle");
            var password = oracle1.Resource.PasswordParameter.Value;
```